### DISCUSSION TITLE
Make the sidebars wider on large screens

### DISCUSSION THREAD

**Discussion by umarcor** [pradyunsg/furo#651] [Ideas] [answered]
Coming from https://github.com/pradyunsg/furo/discussions/536

@pradyunsg would you consider using the additional width on large screens to increase the width of the sidebars but not the body? See the following comparison (current one on the left, suggested on the right):

![image](https://user-images.githubusercontent.com/38422348/222793095-5d73b6e5-9d59-48ef-87ec-06d53abb8e4b.png)

Actually, on my laptop with a 1366px wide resolution, Furo looks closer to the right version than the left. It's on 1920px wide screen where the left one is used.

/cc @KD-7

**Comment by Eric-Arellano** [comment]
An alternative design proposal: keep the sidebars the same size, but allow the page contents itself to grow. Presumably, the page contents are what users are the most interested in. Right now, it's hardcoded to 46em, so there's no user benefit to increasing the width of the site.

See https://github.com/Qiskit/qiskit_sphinx_theme/pull/386 for how we very easily implemented this using Flexbox for our variant of Furo. @pradyunsg I'm happy to open a PR if you like this design.

**Reply by pradyunsg** [reply]
The limited content width is intentional, so that text stays readable even on wide screens. See https://www.mediawiki.org/wiki/Reading/Web/Desktop_Improvements/Features/Limiting_content_width#Goals_and_Motivation for some discussion on why limiting the content width like that is a good thing for prose-heavy webpages.

**Reply by pradyunsg** [reply]
As for whether the sidebar clings to the left edge or the content, that's an aesthetic choice. I picked content in part due to my annoyances with trying to use stuff on wide screen monitors, but it's totally understandable to pick something else.

I'm wary of adding any options to this theme (they increase the surface area by a lot) -- this isn't really a good thing to provide a knob for IMO.

**Comment by pradyunsg** [comment]
> @pradyunsg would you consider using the additional width on large screens to increase the width of the sidebars but not the body?

Yes, that's something that's been on my radar since it was mentioned in that topic! Thanks for opening this separately. :)

**Reply by pradyunsg** [reply]
@umarcor would you be willing to file a trackable task for this? (i.e. a regular Github issue, so that we have a place to discuss the implementation aspects of this, and to avoid a situation where my monkey brain loses track of this!).

**Comment by pradyunsg** [comment, marked as answer]
Ok, looking into this today, Furo currently has only 1 sidebar-width consistently throughout the page widths. This means that the presentation of the sidebars is the same regardless of whether you're on a mobile screen, a laptop screen or an ultra-wide monitor -- I think that's a useful property since it avoids maintainers writing documentation titles that look ugly on screens that aren't the size that they used when authoring things.

Furo has a lot of these little nudges, to make authors experience the documentation's structure as their readers would, and I think leaving the sidebar width constant on all widths is a reasonable thing to do in that context. I was looking into tweaking the whitespace between the content and sidebar, and looks like past me did that already (i.e. that padding has changed since the screenshots in OP). :)

With that, I don't think there's more to do here. Thanks for opening this though! It was a fun exercise to think through this. Lemme know if you have any thoughts/opinions on this. :)

**Reply by Eric-Arellano** [reply]
I agree that it is useful to always have the same sidebar width, for your reasons.

--

FYI, in qiskit-sphinx-theme, we ended up settling on keeping the sidebars always the same width rather than dynamically making them bigger via padding. The main content grows but is capped at 60em per your recommendation in https://github.com/pradyunsg/furo/discussions/651#discussioncomment-6195625. We tried to balance better using real estate on large screens (I develop on an ultrawide monitor) while respecting the guidance you shared.

We also made the left sidebar bigger, 15em to 18em. We have lots of API docs with long page titles, so wanted more real estate for the site ToC. 

See https://github.com/Qiskit/qiskit_sphinx_theme/blob/main/src/qiskit_sphinx_theme/assets/styles/_layout.scss. You can try out the theme at https://qiskit.github.io/qiskit_sphinx_theme/.

The layout has worked out quite well for us!
